### ISSUE TITLE
Add clang (allowing CLANGDWARF and CLANGPDB) to Linux dev containers

### ISSUE THREAD

**Issue by mikebeaton** [tianocore/containers#99]
It seems desirable that the fedora-dev containers include clang (`sudo dnf install -y clang llvm lld`), allowing the CLANGDRAWF and CLANGPDB toolchains to be used. Could this be added?

**Comment by osteffenrh** [comment]
Sure. I'll look into it. The containers need an update anyway.

**Comment by osteffenrh** [comment]
Ubuntu 22 and Fedora 40 dev images now include clang.

**Comment by mikebeaton** [comment]
Thank you very much for that - is there any reason not to do fedora-39 and -37 as well (ideally, or at least -39)? That would be helpful too, if possible.

**Comment by osteffenrh** [comment]
I am planning to remove those images very soon.
The CI is about to get moved over to the F40 image and I would do the same with the devcontainer image too.

It is just too much work to maintain all those images.

The images were initially intended mainly for the CI and this is where my personal focus is.

Would it be ok for you to use the F40 image instead of the other Fedora ones?

**Comment by mikebeaton** [comment]
Yes, esp. edk2 main image is moving over shortly - no problem - tyvm

**Comment by mikebeaton** [comment]
@osteffenrh -

Hi again!

I just looked more carefully at the different images. IMO it would be very useful to add CLANGDWARF and CLANGPDB CI. I work on a fork of edk2 which runs CI in XCODE5, VS2019, GCC, CLANGDWARF and CLANGPDB - different toolchains often trap different real errors.

So - it would make most sense to me if clang installs where added to the `-build` containers (from which I understand the -dev containers inherit), if at all possible? (I was actually hoping to make a PR for CLANGDWARF and CLANGPDB CI.)
